src/nerpa_types.py

```diff
@@ -0,0 +1,138 @@
+from __future__ import annotations
+from typing import (
+    Dict,
+    List,
+    Tuple)
+from enum import auto, Enum
+import yaml
+from dataclasses import asdict, dataclass
+
+LProb = float  # logarithm of probability
+MonomerResidue = str
+
+
+def enum_representer(dumper, e: Enum):
+    return dumper.represent_scalar(f'!{e.__class__.__name__}', e.name)
+
+
+class Chirality(Enum):
+    L = auto()
+    D = auto()
+    Unknown = auto()
+
+yaml.add_representer(Chirality, enum_representer)
+
+
+class PTM(Enum):  # post-translational modification
+    Methylation = auto()
+    Unknown = auto()
+
+yaml.add_representer(PTM, enum_representer)
+
+
+@dataclass
+class NRP_Monomer:
+    residue: MonomerResidue
+    ptms: Tuple[PTM, ...]
+    chirality: Chirality
+    rBAN_name: str
+
+
+class BGC_Module_Modification(Enum):
+    Epimerization = auto()
+    Methylation = auto()
+
+yaml.add_representer(BGC_Module_Modification, enum_representer)
+
+
+@dataclass
+class BGC_Module_Prediction:
+    gene_id: str
+    module_idx: int
+    residue_score: Dict[MonomerResidue, LProb]
+    modifications: Tuple[BGC_Module_Modification, ...]
+    iterative_module: bool
+    iterative_gene: bool
+
+
+@dataclass
+class BGC_variant:
+    genome_id: str
+    bgc_id: str
+    predictions: List[BGC_Module_Prediction]
+
+
+@dataclass
+class NRP_fragment:
+    monomers: List[NRP_Monomer]
+    is_cyclic: bool
+    to_rban_indexes: List[int]
+
+
+@dataclass
+class NRP_variant:
+    nrp_name: str
+    fragments: List[NRP_fragment]
+
+
+def main():
```

Comment:
Let's move it outside this file to keep it clean. Instead, you can create a separate Python file (e.g., `test_data_types.py`) that imports this file (akin C++ header) and does all these fancy test things.

Reply:
Sure. `main` is there only for demonstration purposes, so that you know how to generate yaml files (I recall you planned to).

Reply:
Yep, and it is good to have such a demonstration thing. I just found having it in another file is more convenient :)